conans/test/integration/toolchains/cmake/cmakedeps/test_cmakedeps.py

```diff
@@ -707,3 +707,35 @@ def generate(self):
     assert 'set(dep_NO_SONAME_MODE_RELEASE TRUE)' in dep
     other = c.load("app/other-release-data.cmake")
     assert 'set(other_other_mycomp1_NO_SONAME_MODE_RELEASE TRUE)' in other
+
+def test_cmakedeps_set_legacy_variable_name():
+    client = TestClient()
+    conanfile = str(GenConanfile("dep", "1.0"))
```

Comment:
You could call this `base_conanfile` here and avoid calling `GenConanfile` twice.